src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java

```diff
@@ -0,0 +1,32 @@
+package com.endava.internship.mocking.repository;
+
+import com.endava.internship.mocking.model.Status;
+import com.endava.internship.mocking.model.User;
+import com.endava.internship.mocking.repository.InMemUserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class InMemPaymentRepositoryTest {
+
+    InMemUserRepository userRepository;
+
+    @BeforeEach
+    void init() {
+        userRepository = new InMemUserRepository();
+    }
+
+    @Test
+    void findById() {
+        User expectedUser = new User(1, "John", Status.ACTIVE);
+
+        Exception exception = assertThrows(IllegalArgumentException.class, () -> userRepository.findById(null));
+        assertEquals(exception.getMessage(), "User id must not be null");
+
+        assertEquals(userRepository.findById(0), Optional.empty());
+        assertEquals(userRepository.findById(1), Optional.of(expectedUser));
```

Comment:
Good test, just one comment: you try to check here 2 different test cases: negative and positive. Usually for every case we create separate test. Do it, please, and give to tests verbose names, don't worry, name can be very long.

Reply:
done